### PR TITLE
Feature/issue 31 wait after trick

### PR DESCRIPTION
**Pull request by creisle**
New Features
- adds 'ready for trick' checkin event to wait before dealing the next trick

### CODE REVIEW

File: server/db/round.ts [hunk 1]
```diff
@@ -163,20 +164,31 @@ const playCard = async (redis: Redis, gameId: string, playerId: string, cardSuit
     // if this was the last card in the trick, evaluate the trick
     const trickComplete = Boolean(trickCards.length + 1 === players.length);
     let trickWinner = null;
-    const roundComplete = trickComplete && trickNumber === roundNumber;
     if (trickComplete) {
         trickWinner = await evaluateTrick(redis, gameId, roundNumber, trickNumber);
-
-        if (!roundComplete) {
```
Comment (areisle) [inline server/db/round.ts:170]: why did you remove all of this?

File: server/index.ts [hunk 2]
```diff
@@ -226,6 +206,61 @@ const server = async ({ port = 3000 }) => {
...
+                    if (!roundComplete) {
+                        await db.setCurrentTrick(redis, gameId, trickNumber + 1);
+                    } else {
+                        await Promise.all([
+                            db.setCurrentTrick(redis, gameId, 0),
+                            db.setCurrentRound(redis, gameId, roundNumber + 1),
+                        ]);
+                    }
+
+                    if (roundComplete) {
```
Comment (areisle) [inline server/index.ts]: why not combine this with the if/else above?

File: src/types.ts [hunk 3]
```diff
@@ -72,24 +72,26 @@ export interface RejoinGameParams extends Partial<GameState> {
...
     REJOIN_GAME = 'rejoin-game',
-    GET_PLAYERS = 'get-users',
+    START_GAME = 'start-game',
 }
 
 export enum SERVER_EVENTS {
-    TRUMP_CHANGED = 'trump-changed',
     ACTIVE_PLAYER_CHANGED = 'active-user-changed',
-    TRICK_WON = 'trick-won',
+    ALL_PLAYERS_READY = 'all-players-ready',
```
Comment (areisle) [inline src/types.ts]: I think this isn't used?

**Review by areisle — CHANGES_REQUESTED** [review]

File: src/types.ts (hunk 3 above)
Comment (creisle) [reply src/types.ts]: yup will remove

File: server/db/round.ts (hunk 1 above)
Comment (creisle) [reply server/db/round.ts:170]: because this used to happen when the last card was played, but now it happens when every player has said ready

File: server/index.ts (hunk 2 above)
Comment (creisle) [reply server/index.ts]: sure i can do that.  I just moved if from the event above is why it was left like this

**Comment by creisle** [comment]
@areisle I think i found a bug, when a new trick started the leader was for the current trick and not the upcoming one (24532e5)

**Review by areisle — APPROVED** [review]
